payments-core/src/main/java/com/stripe/android/paymentsheet/analytics/DefaultEventReporter.kt

```diff
@@ -1,40 +1,23 @@
 package com.stripe.android.paymentsheet.analytics
 
-import android.content.Context
-import com.stripe.android.PaymentConfiguration
 import com.stripe.android.networking.AnalyticsRequestExecutor
 import com.stripe.android.networking.AnalyticsRequestFactory
-import com.stripe.android.networking.DefaultAnalyticsRequestExecutor
+import com.stripe.android.payments.core.injection.IOContext
 import com.stripe.android.paymentsheet.PaymentSheet
 import com.stripe.android.paymentsheet.model.PaymentSelection
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import javax.inject.Inject
 import kotlin.coroutines.CoroutineContext
 
-internal class DefaultEventReporter internal constructor(
+internal class DefaultEventReporter @Inject internal constructor(
```

Comment:
The `@Singleton` annotation itself doesn't do anything since we're not using a `@Singleton` component. `DefaultEventReporter` can actually exist in multiple instances if someone integrates both custom and complete paths, like the `paymentsheet-example` app.

Reply:
https://stackoverflow.com/a/44237507/849963

Reply:
`@Singleton` constructor doesn't need to take effect in a `@Singleton` component, I actually just verified it with the follows: 
* add two break points [one](https://github.com/stripe/stripe-android/blob/cdd2def71f8732c5ccf164a51ccf2501335dab06/payments-core/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt#L125) and [two](https://github.com/stripe/stripe-android/blob/453499e8fb15043279f831112b8f5d9560b65fe7/payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt#L142)
1. without `@Singleton` annotated, open paymentsheet custom, check the address of `DefaultFlowController.eventReporter` in one, then try change a payment method, check the address of `PaymentOptionsViewModel.Factory.eventReporter` in two, you'll found they are two difference instances
2. with `@Singleton` annotated, do the same check, they are the same instance

Usually the scope annotate is added to a component to make sure its subcomponent's scope is smaller than it. Since here we just have one component instance created for complete path(`PaymentSheetViewModelComponent`) and custom path(`FlowControllerComponent`) without any subcomponent that lives shorter, it doesn't matter we annotate this component instance with an annotation or not.

Reply:
Hmm, that is correct, it makes the object singleton within the Component. Thanks for explaining!